src/expr.rs

```diff
@@ -315,12 +305,12 @@ pub fn format_expr(
         })
 }
 
+pub struct PairParts<'a>(pub &'a str, pub &'a str, pub &'a str);
+
 pub fn rewrite_pair<LHS, RHS>(
     lhs: &LHS,
     rhs: &RHS,
-    prefix: &str,
-    infix: &str,
-    suffix: &str,
+    pp: &PairParts,
```

Comment:
I suppose you are trying to suppress needless_take_by_value lint with this useless reference... Consider adding `#[derive(Copy, Clone)]` to `PairParts` definition, and change the argument type to just `PairParts`. That lint ignores `Copy` types.